service/src/java/org/apache/hive/service/cli/thrift/ThriftCLIService.java

```diff
@@ -147,15 +150,80 @@
 
   protected ThreadLocal<ServerContext> currentServerContext;
 
+  /**
+   * A context container for information related to client connections to the
+   * Thrift server.
+   */
   static class ThriftCLIServerContext implements ServerContext {
-    private SessionHandle sessionHandle = null;
 
-    public void setSessionHandle(SessionHandle sessionHandle) {
-      this.sessionHandle = sessionHandle;
+    private final AtomicInteger messagesProcessedCounter;
```

Comment:
could we replace AtomicInteger with a integer val here, though it works here?

Reply:
Ya, I think it's OK to do that.